framework/material/Dialog.kt

```diff
@@ -49,11 +50,22 @@ object Dialog {
         state: DialogState, title: String, width: Dp, height: Dp,
         padding: Dp = Theme.DIALOG_PADDING,
         content: @Composable DialogWindowScope.() -> Unit
-    ) = Dialog(
-        title = title, onCloseRequest = { state.close() }, state = rememberDialogState(
+    ) = Layout(state, title, padding,
+        rememberDialogState(
             position = WindowPosition.Aligned(Alignment.Center),
             size = DpSize(width, height)
-        )
+        ),
+        content
+    )
+
+    @Composable
+    fun Layout(
+        state: DialogState, title: String,
+        padding: Dp = Theme.DIALOG_PADDING,
+        composeDialogState: ComposeDialogState,
+        content: @Composable DialogWindowScope.() -> Unit
+    ) = Dialog(
```

Comment:
By allowing the option to pass in a `ComposeDialogState` directly rather than always constructing it based on the `width` and `height`, we allow the parent the option of controlling the size of dialog more directly.